ee/presencedetection/presencedetection.go

```diff
@@ -0,0 +1,116 @@
+package presencedetection
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kolide/launcher/ee/consoleuser"
+)
+
+type PresenceDetectionResponse struct {
+	Success bool   `json:"success"`
+	Error   string `json:"error,omitempty"`
+}
+
+type PresenceDetector struct {
+	lastDetectionUTC time.Time
+	mutext           sync.Mutex
+}
+
+func (pd *PresenceDetector) DetectForConsoleUser(reason string, detectionInterval time.Duration) (bool, error) {
```

Comment:
Is some of this going to move inside the desktop runner instead, now that desktop is always running?

Reply:
Yea, will drop the whole exec stuff an move the call to DesktopRunner. Still trying to find some elegant way for LocalServer to ask DesktopRunner to to presence detection.